### ISSUE TITLE
Can't create database with navicat: Dolt does not accept single quotes on charset identifier 

### ISSUE THREAD

**Issue by Jinsns** [dolthub/dolt#4977]
![image](https://user-images.githubusercontent.com/56482300/207387739-7db50495-a0f1-42d4-8244-5da410bec512.png)

I'm using navicat premium 15. When I try to create a new database or import data from a csv to create a new table, error happens. 
I guess there are some problems with the quotation mark in the sql corresponding to the operation of navicat. If I copy that sql to cli, same error will happen, while ok without the quotation mark.
![image](https://user-images.githubusercontent.com/56482300/207388511-c9b59579-4c8e-4907-99c2-7b4c76badf77.png)

**Comment by timsehn** [comment]
So, these queries work on the tip of main outside of navicat:
```
$ dolt sql-client
--user or -u argument is required
$ dolt sql-client -u root
# Welcome to the Dolt MySQL client.
# Statements must be terminated with ';'.
# "exit" or "quit" (or Ctrl-D) to exit.
mysql> create database `foo`
    -> ;
mysql> create database `bar` character set `utf8mb4`;
mysql> show databases;
+--------------------+
| Database           |
+--------------------+
| bar                |
| foo                |
| information_schema |
| mysql              |
+--------------------+
4 rows in set (0.00 sec)

mysql>
```

I think we just added support for collations on databases on Nov 1. 

https://github.com/dolthub/dolt/pull/4629

Are you using the latest Dolt version?

**Comment by fulghum** [comment]
Thanks for reporting this! I've made a quick change to our sql parser to allow charsets and collation names to be in single quotes or double quotes when used in `create database` or `alter database` statements. I took a quick look at other places charsets/collation names are used (e.g. create table options) and it looks like single/double quotes are already supported there. 

We'll get this merged in today and out in the next release of Dolt, likely today or tomorrow. Using backtick quotes is a good workaround for this until the fix is released.
